src/vehicle_use.cpp

```diff
@@ -2118,6 +2118,54 @@ void vehicle::build_interact_menu( veh_menu &menu, const tripoint &p, bool with_
         } );
     }
 
+    // Remove attached tools that have become incompatible with workstation because of migration etc
+    if( vp.avail_part_with_feature( "VEH_TOOLS" ) && !vp.get_tools().empty() ) {
+        const std::optional<vpart_reference> vp_toolstation = vp.avail_part_with_feature( "VEH_TOOLS" );
+        if( vp_toolstation->info().toolkit_info ) {
```

Comment:
We actually already have a `vp_toolstation` in this scope later in the file around line 2352, because that's where parts of this are copypasted from :grin: This check was originally located there, but we need it to precede the  line 2169-starting block because obsolete tools need to be cleared before the "use tool x" menu entries are built.

Reply:
So it's possible that by popping out all the tools, later on there won't be any more `avail_part_with_feature( "VEH_TOOLS" )`? Gotcha.

Reply:
No, VEH_TOOLS isn't provided by a tool in the vehicle part. I'm just bad at explaining :laughing:

The pattern used elsewhere in the file is that we introduce a vpart_reference variable of vp.avail_prt_with_feature("THE_THING_WE_WANT_TO_ADD_TO_THE_MENU"), and this is done later in the file for VEH_TOOLS where the actual menu entries for adding and removing tools from the workstation are created. But I think it's fine to diverge from that pattern here and pull the variable here already. I've done that in the latest commit.